modules/run-vault/run-vault

```diff
@@ -112,16 +115,34 @@ function generate_vault_config {
   local readonly api_addr="$5"
   local readonly config_dir="$6"
   local readonly user="$7"
-  local readonly enable_s3_backend="$8"
-  local readonly s3_bucket="$9"
-  local readonly s3_bucket_region="${10}"
+
+  if [[ "$enable_s3_backend" == "true" ]]; then
+    local readonly enable_s3_backend="$8"
+    local readonly s3_bucket="$9"
+    local readonly s3_bucket_region="${10}"
+    local readonly enable_dynamo=""
+    local readonly dynamo_region=""
+    local readonly dynamo_table=""
+  fi
+
+  if [[ "$enable_dynamo" == "true" ]]; then
+    local readonly enable_dynamo="$8"
+    local readonly dynamo_region="$9"
+    local readonly dynamo_table="${10}"
+    local readonly enable_s3_backend=""
+    local readonly s3_bucket=""
+    local readonly s3_bucket_region=""
+  fi
```

Comment:
I find changing the list of input params a bit confusing. I'd recommend making this a single list:

```bash
local readonly enable_s3_backend="$8"
local readonly s3_bucket="$9"
local readonly s3_bucket_region="${10}"
local readonly enable_dynamo="${11}"
local readonly dynamo_region="${12}"
local readonly dynamo_table="${13}"
```

The `run` method should pass all these params, setting them to true, false, empty strings, or proper values, depending on what the user passed in.

Reply:
I agree, however, by default the vault config generation just adds the S3 params at the end (no harm if they are empty).  I can add extra vault config generation logic (at the bottom) if you'd rather see this as a list.

Reply:
Changed it back to a list and updated logic at the bottom (vault generation script) to match